pkg/controller/ibmblockcsi/ibmblockcsi_controller.go

```diff
@@ -493,6 +464,52 @@ func (r *ReconcileIBMBlockCSI) reconcileServiceAccount(instance *ibmblockcsi.IBM
 			if err != nil {
 				return err
 			}
+
+			controllerStatefulset, err := r.getControllerStatefulSet(instance)
+			if err != nil {
+				return err
+			}
+
+			nodeDaemonSet, err := r.getNodeDaemonSet(instance)
+			if err != nil {
+				return err
+			}
+
+			if controllerServiceAccountName == sa.Name {
+				controllerlogger := log.WithValues("Resource Type", "Controller")
+				controllerPod := &corev1.Pod{}
+				err := r.getControllerPod(controllerStatefulset, controllerPod)
+				if errors.IsNotFound(err) {
+					return nil
```

Comment:
will this prevent from the second service account to be reconciled?

Reply:
I think maybe I will add the `if errors.IsNotFound(err)` in the `getControllerPod`, and remove it from checking here and in the other place that we call to `getControllerPod` function. What do you think?

Reply:
it would still require every caller to check whether the pod is not nil, but I guess it's better than nothing?
I think here specifically the caller doesn't need the pod, so it would be better if it could call a function that doesn't require the pod as a parameter

Reply:
done